packages/shared/src/hooks/notifications/useNotificationPreferenceToggle.ts

```diff
@@ -0,0 +1,69 @@
+import { useMutation } from '@tanstack/react-query';
+import { useMemo } from 'react';
+import { NotificationPreferenceStatus } from '../../graphql/notifications';
+import {
+  useNotificationPreference,
+  checkHasStatusPreference,
+  UseNotificationPreferenceProps,
+} from './useNotificationPreference';
+
+export type UseNotificationPreferenceToggleProps = {
+  params: UseNotificationPreferenceProps['params'][0] | undefined;
+};
+
+export type UseNotificationPreferenceToggle = {
+  isSubscribed: boolean;
+  isReady: boolean;
+  onToggle: () => Promise<{ isSubscribed: boolean }>;
+};
+
+export const useNotificationPreferenceToggle = ({
+  params,
+}: UseNotificationPreferenceToggleProps): UseNotificationPreferenceToggle => {
+  const {
+    preferences,
+    subscribeNotification,
+    clearNotificationPreference,
+    isFetching,
+    isPreferencesReady,
+  } = useNotificationPreference({
+    params: params ? [params] : undefined,
+  });
+  const isSubscribed = useMemo(() => {
+    return !!preferences?.some((item) =>
+      checkHasStatusPreference(
+        item,
+        params?.notificationType,
+        params?.referenceId,
+        [NotificationPreferenceStatus.Subscribed],
+      ),
+    );
+  }, [preferences, params?.notificationType, params?.referenceId]);
+
+  const { mutateAsync: onToggle } = useMutation({
```

Comment:
Can we call it something else than `onToggle`? E.g. even simple `toggle` would probably be better, since we are actually mutating the state here, not just reacting to toggle event in the UI.

Reply:
We usually call handlers with this kind of nomenclature. Don't want to change at this time if that is ok?